framework/logging.cpp

```diff
@@ -2483,17 +2485,20 @@ void YamlLogger::print()
     }
 
     // print the thread messages
-    auto doprint = [this](PerThreadData::Common *data, int i) {
+    auto doprint = [this, init_skip_message_bytes](PerThreadData::Common *data, int s_tid) {
         struct mmap_region r = maybe_mmap_log(data);
 
-        if (r.size == 0 && !data->has_failed() && sApp->shmem->verbosity < 3)
+        /* for main threads (negative ids) adjust the message size to account
+         * for skip message. this is to avoid empty messages printed to the log.
+         */
+        if (r.size - (s_tid < 0 ? init_skip_message_bytes : 0) == 0 && !data->has_failed() && sApp->shmem->verbosity < 3)
             return;             /* nothing to be printed, on any level */
 
-        print_thread_header(file_log_fd, i, INT_MAX);
+        print_thread_header(file_log_fd, s_tid, INT_MAX);
         int lowest_level = print_one_thread_messages(file_log_fd, r, INT_MAX);
 
         if (lowest_level <= sApp->shmem->verbosity && file_log_fd != real_stdout_fd) {
-            print_thread_header(real_stdout_fd, i, sApp->shmem->verbosity);
+            print_thread_header(real_stdout_fd, s_tid, sApp->shmem->verbosity);
             print_one_thread_messages(real_stdout_fd, r, sApp->shmem->verbosity);
         }
 
```

Comment:
If you return when `r.size != 0`, then `munmap_and_truncate_log` doesn't truncate the log. I wonder if the existing second part of the conditional is a latent bug too.